CMakeLists.txt

```
@@ -16,11 +16,64 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(rosidl_default_generators REQUIRED)
 
+set(protobuf_MODULE_COMPATIBLE TRUE)
+find_package(Protobuf CONFIG REQUIRED)
+message(STATUS "Using protobuf ${Protobuf_VERSION}")
+
+set(_PROTOBUF_LIBPROTOBUF protobuf::libprotobuf)
+set(_REFLECTION gRPC::grpc++_reflection)
+
+find_package(gRPC CONFIG REQUIRED)
+message(STATUS "Using gRPC ${gRPC_VERSION}")
+
+# Generated sources
+add_custom_command(
+      OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/akushon.pb.cc" 
+            "${CMAKE_CURRENT_BINARY_DIR}/akushon.pb.h"
+            "${CMAKE_CURRENT_BINARY_DIR}/akushon.grpc.pb.cc"
+            "${CMAKE_CURRENT_BINARY_DIR}/akushon.grpc.pb.h"
+      COMMAND $<TARGET_FILE:protobuf::protoc>
+      ARGS --grpc_out "${CMAKE_CURRENT_BINARY_DIR}"
+        --cpp_out "${CMAKE_CURRENT_BINARY_DIR}"
+        -I ${CMAKE_CURRENT_SOURCE_DIR}/proto
+        --plugin=protoc-gen-grpc=$<TARGET_FILE:gRPC::grpc_cpp_plugin>
+        ${CMAKE_CURRENT_SOURCE_DIR}/proto/akushon.proto
+      DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/proto/akushon.proto)
+
+add_library(akushon_proto
+    ${CMAKE_CURRENT_BINARY_DIR}/akushon.pb.cc
+    ${CMAKE_CURRENT_BINARY_DIR}/akushon.pb.h
+    ${CMAKE_CURRENT_BINARY_DIR}/akushon.grpc.pb.cc
+    ${CMAKE_CURRENT_BINARY_DIR}/akushon.grpc.pb.h)
+
+target_compile_options(akushon_proto PRIVATE -fPIC)
+set_property(TARGET akushon_proto PROPERTY CXX_STANDARD 17)
```

I'm sorry for the wrong suggestion, but setting C++ standard using [`target_compile_features`](https://cmake.org/cmake/help/latest/command/target_compile_features.html) seems is better than setting it using `set_property`.

```suggestion
target_compile_features(akushon_proto PUBLIC cxx_std_17)
```